lib/astarte/utilities/params_gen.ex

```diff
@@ -0,0 +1,175 @@
+
+  ## Usage Examples
+
+  ### Example
+
+  In this example, we override the default `a` integer generator with a tuple {2, 3, 4}.
+
+  defmodule MyGenerators do
+    use Astarte.Generators.Utilities.ParamsGen
+
+    # Override the default integer generator using params gen all
+    def parametric_generators(params \\ [a: { 2, 3, 4}])
+    params gen(
+      all a <- integer(),
+          b <- list_of(string(:ascii)),
+          c <- constant({:amicizia, "dottore"}),
+          params: params
+    ) do
+        {a, b, c}
+      end
+  end
+
+  ## Notes
+
+  - **Integration with ExUnitProperties:** This macro leverages the existing functionality of ExUnitProperties,
+    making it easy to adopt if you are already using property-based testing in your project.
+  - **Macro Syntax:** The macro expects a keyword list under the `params:` key, where each key corresponds to
+    a generator name (e.g., `a`, `b`) and each value is the custom generator or fixed params to be used.
+  - **Fallback Behavior:** For generators not specified in the override list, the macro will default to using
+    the original generator from ExUnitProperties.
+  - **Compile-Time Verification:** Misuse or incorrect configuration will be flagged at compile time, helping
+    you catch errors early in the development process.
+
+  """
+
+  @doc """
+  Injects the necessary imports to use ParamsGen functionalities.
+  This macro brings in the current module, StreamData, and ExUnitProperties, which are required for property-based tests with custom generator overrides.
+  """
+  @spec __using__(opts :: any()) :: Macro.t()
+  defmacro __using__(_opts) do
+    quote do
+      import unquote(__MODULE__)
+      import StreamData
+      import ExUnitProperties
+    end
+  end
+
+  @doc false
+  @spec params({:gen, any(), [{:all, any(), any()}]}) :: Macro.t()
+  defmacro params({:gen, _gen_meta, [{:all, _all_meta, clauses_with_body}]}) do
+    {clauses, [[do: body]]} = Enum.split(clauses_with_body, -1)
+    compile(clauses, body, __CALLER__)
+  end
+
+  @doc """
+  Defines a custom property-based test generator macro (`params gen all`) that supports overriding default generators.
+  It processes generator clauses and applies any custom overrides provided via the `:params` keyword.
+  """
+  @spec params({:gen, any(), [{:all, any(), list()}]}, [{:do, any()}]) :: Macro.t()
+  defmacro params({:gen, _gen_meta, [{:all, _all_meta, clauses}]}, do: body) do
+    compile(clauses, body, __CALLER__)
+  end
+
+  @streamdata_functions StreamData.__info__(:functions) |> Enum.map(&elem(&1, 0))
+  defp stream_data?(_aliases, {func, _func_meta, _}) when func in @streamdata_functions,
+    do: true
+
+  defp stream_data?(
+         aliases,
+         {{:., _dot_meta, [{:__aliases__, _aliases_meta, alias_list}, func]}, _other_meta, _rest}
+       )
+       when func in @streamdata_functions do
+    Module.concat(alias_list) in aliases
+  end
+
+  defp stream_data?(_, _), do: false
```

Comment:
If we change this function like so

```suggestion
  defp stream_data?(overridden_value, env) do
    {overridden_value, _bindings} = Code.eval_quoted(overridden_value, [], env)
    is_struct(overridden_value, StreamData)
  end
```